### PR TITLE
fix: clean log should use transport options file

### PR DESCRIPTION
**Pull request by denghongcai**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

clean_log

##### Description of change

clean_log should use transport options as input

### CODE REVIEW

**Comment by popomore** [comment]
补充一个 logger 带多个 FileTransport 的用例

**Comment by popomore** [comment]
ci 挂了

**Comment by denghongcai** [comment]
> ci 挂了

重试了下 ci 过了

**Comment by popomore** [comment]
+ egg-logrotator@3.0.6

File: app/schedule/clean_log.js [hunk 1]
```diff
@@ -13,20 +13,25 @@ module.exports = app => ({
...
+          logDirs.add(logDir);
         }
       }
     }
 
     const maxDays = app.config.logrotator.maxDays;
     if (maxDays && maxDays > 0) {
       try {
-        const tasks = logDirs.map(logdir => removeExpiredLogFiles(logdir, maxDays, logger));
+        const tasks = Array.from(logDirs, logdir => removeExpiredLogFiles(logdir, maxDays, logger));
```
Comment (atian25) [inline app/schedule/clean_log.js]: ^_^ 原来 Array.from 还有第二个参数啊

**Pull request merged**